apps/emqx_durable_storage/src/emqx_ds_int.hrl

```diff
@@ -17,11 +17,18 @@
 -define(EMQX_DS_HRL, true).
 
 -define(SESSION_TAB, emqx_ds_session).
+-define(ITERATOR_REF_TAB, emqx_ds_iterator_ref).
 -define(DS_SHARD, emqx_ds_shard).
 
 -record(session, {
     id :: emqx_ds:session_id(),
     iterators :: #{emqx_topic:words() => emqx_ds:iterator_id()}
 }).
 
+-record(iterator_ref, {
```

Comment:
We changed to using a table for the references based on this: https://github.com/emqx/emqx/pull/11362#discussion_r1289244353
Since an increasing number of subscription would result in larger and larger maps being read then written back to storage.